src/components/Store/Store.js

```diff
@@ -75,24 +81,63 @@ const Store = () => {
   const clearFilter = () => {
     setDisplayedGames(games);
     setFilter('none');
+    setQuery('');
+  };
+
+  const changeDisplay = (display) => {
+    setDisplay(display);
+  };
+
+  const handleSearch = () => {
+    const searchedGames = [];
+    for (let game of games) {
+      if (game.name.toLocaleLowerCase().includes(query)) {
+        searchedGames.push(game);
+      }
+    }
+
+    setDisplayedGames(searchedGames);
+  };
+
+  const handleQuery = (message) => {
+    if (message.length === 0) setDisplayedGames(games);
+    setQuery(message);
   };
 
   return (
     <div className={styles.store}>
-      <Nav leftButtons={leftNavButtons} rightButtons={rightNavButtons} />
+      <Nav
+        leftButtons={leftNavButtons}
+        rightButtons={rightNavButtons}
+        handleSearch={handleSearch}
+        handleQuery={handleQuery}
+        query={query}
+      />
       <div className={styles.content}>
         <StoreNav filterBy={filterBy} currentFilter={filter} />
         <div className={styles['mid-content']}>
           <h2 className={styles.title}>Trending and interesting</h2>
           <span>Based on player counts and ratings</span>
-          <FilterNav filter={filter} clearFilter={clearFilter} />
-          <div className={styles.collection}>
+          <FilterNav
+            filter={filter}
+            clearFilter={clearFilter}
+            changeDisplay={changeDisplay}
+            display={display}
+          />
+          <div
```

Comment:
Since this has become a bit more complex I would extract this logic in a separate `GameList` component, which receives, 3 props `games={displayedGames}` `wishList={wishList}` and `handleWishList`. I would keep the the `display` state inside that component since you don't need it anywhere else. The `styles` for displaying either `grid` or `row` should also be moved there.